apps/aeutil/lib/bits.ex

```diff
@@ -1,15 +1,45 @@
 # Taken from http://minhajuddin.com/2016/11/01/how-to-extract-bits-from-a-binary-in-elixir/
 # License: CC BY-SA 3.0
 defmodule Aeutil.Bits do
-  def bech32_encode(prefix, bin) do
-    SegwitAddr.encode(prefix, 0, :binary.bin_to_list(bin))
+  def encode58c(type, payload) when is_binary(payload) do
+    prefix =
+      case type do
```

Comment:
my imagination was to use the atoms in both ways, like this it makes no sense, you can pass the prefix from the calling function

Reply:
I thought we use the atom as just to clarify the prefix that we are using. After all we have to get the string value of the prefix. Yes, we could just pass it in the calling function, but maybe this way is more readable. That was our understanding of the your idea. Can u elaborate a bit futher on this, maybe we are missing the point for some reason.

Reply:
for decoding you pattern match the prefix - which is fine, my idea was to pass the expected prefix as atom

Reply:
Oh, and then convert it to string ?

Reply:
then pattern match both ways to/from string. But your approach is fine, just pass the prefix as string instead of atom, to match the decoding.